src/topology/algebra/group.lean

```diff
@@ -1,105 +1,241 @@
+
+variables [topological_space α] [group α]
+
+@[to_additive continuous_neg']
+lemma continuous_inv' [topological_group α] : continuous (λx:α, x⁻¹) :=
+topological_group.continuous_inv α
+
+@[to_additive continuous_neg]
+lemma continuous_inv [topological_group α] [topological_space β] {f : β → α}
+  (hf : continuous f) : continuous (λx, (f x)⁻¹) :=
+hf.comp continuous_inv'
+
+@[to_additive tendsto_neg]
+lemma tendsto_inv [topological_group α] {f : β → α} {x : filter β} {a : α}
+  (hf : tendsto f x (nhds a)) : tendsto (λx, (f x)⁻¹) x (nhds a⁻¹) :=
+hf.comp (continuous_iff_continuous_at.mp (topological_group.continuous_inv α) a)
+
+@[to_additive prod.topological_add_group]
+instance [topological_group α] [topological_space β] [group β] [topological_group β] :
+  topological_group (α × β) :=
+{ continuous_inv := continuous.prod_mk (continuous_inv continuous_fst) (continuous_inv continuous_snd) }
+
+attribute [instance] prod.topological_add_group
+
+protected def homeomorph.mul_left [topological_group α] (a : α) : α ≃ₜ α :=
+{ continuous_to_fun  := continuous_mul continuous_const continuous_id,
+  continuous_inv_fun := continuous_mul continuous_const continuous_id,
+  .. equiv.mul_left a }
+attribute [to_additive homeomorph.add_left._proof_1] homeomorph.mul_left._proof_1
+attribute [to_additive homeomorph.add_left._proof_2] homeomorph.mul_left._proof_2
+attribute [to_additive homeomorph.add_left._proof_3] homeomorph.mul_left._proof_3
+attribute [to_additive homeomorph.add_left._proof_4] homeomorph.mul_left._proof_4
+attribute [to_additive homeomorph.add_left] homeomorph.mul_left
+
+@[to_additive is_open_map_add_left]
+lemma is_open_map_mul_left [topological_group α] (a : α) : is_open_map (λ x, a * x) :=
+(homeomorph.mul_left a).is_open_map
+
+protected def homeomorph.mul_right
+  {α : Type*} [topological_space α] [group α] [topological_group α] (a : α) :
+  α ≃ₜ α :=
+{ continuous_to_fun  := continuous_mul continuous_id continuous_const,
+  continuous_inv_fun := continuous_mul continuous_id continuous_const,
+  .. equiv.mul_right a }
+attribute [to_additive homeomorph.add_right._proof_1] homeomorph.mul_right._proof_1
+attribute [to_additive homeomorph.add_right._proof_2] homeomorph.mul_right._proof_2
+attribute [to_additive homeomorph.add_right._proof_3] homeomorph.mul_right._proof_3
+attribute [to_additive homeomorph.add_right._proof_4] homeomorph.mul_right._proof_4
+attribute [to_additive homeomorph.add_right] homeomorph.mul_right
+
+@[to_additive is_open_map_add_right]
+lemma is_open_map_mul_right [topological_group α] (a : α) : is_open_map (λ x, x * a) :=
+(homeomorph.mul_right a).is_open_map
+
+protected def homeomorph.inv (α : Type*) [topological_space α] [group α] [topological_group α] :
+  α ≃ₜ α :=
+{ continuous_to_fun  := continuous_inv',
+  continuous_inv_fun := continuous_inv',
+  .. equiv.inv α }
+attribute [to_additive homeomorph.inv._proof_1] homeomorph.inv._proof_1
```

Comment:
Shouldn't this be `to_additive homeomorph.neg._proof_1`? Idem on the next two lines.

Reply:
I think so (although this has nothing to do with this PR)